### ISSUE TITLE
Implement Interface doesn't handle ref returns

### ISSUE THREAD

**Issue by dpoeschl** [dotnet/roslyn#14131]
**Version Used**: preview/25725

Given an interface

``` C#
interface I
{
    ref int M();
}
```

Implement interface produces

``` C#
class C : I
{
    public int M()
    {
        throw new NotImplementedException();
    }
}
```

which is incorrect and does not compile.

**Comment by dpoeschl** [comment]
Probable dupe of #8438

**Comment by VSadov** [comment]
the code simply misses 'ref', while annoying, user can workaround it by adding ref

can be fixed in 2.1

**Comment by jcouv** [comment]
Looks like this was fixed already (see screenshot from 15.6 preview 6):
![image](https://user-images.githubusercontent.com/12466233/36646652-56c7068a-1a2f-11e8-937e-bf0e50c1485a.png)
